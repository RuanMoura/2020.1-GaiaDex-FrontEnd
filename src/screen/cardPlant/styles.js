import { StyleSheet, Dimensions } from 'react-native';
import { gray, green, purple } from '../../theme/colorPalette';

const { height, width } = Dimensions.get('screen');

const styles = StyleSheet.create({
  containerView: {
    flexDirection: 'column',
    backgroundColor: gray.iron(),
    width,
    // backgroundColor: gray.shark(),
    flex: 1,
  },
  containerHeader: {
    flex: 0.1,
    backgroundColor: gray.shark(),
    width,
    alignSelf: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  headerIcons: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  containerPlant: {
    flex: 0.3,
    backgroundColor: gray.shark(),
    flexDirection: 'column',
  },
  plantImg: {
    flex: 0.4,
    width: width / 2.1,
    alignSelf: 'center',
    marginTop: '-9%',
  },
  plantInfo: {
    backgroundColor: gray.shark(),
    width: width / 1.3,
    flex: 0.25,
    flexDirection: 'column',
    alignItems: 'flex-start',
    alignSelf: 'center',
  },
  plantInfoTitle: {
    flex: 0.6,
    color: gray.iron(),
    fontSize: 23,
  },
  plantInfoStats: {
    flex: 0.4,
    flexDirection: 'row',
    width: width / 1.3,
    justifyContent: 'space-between',
    alignContent: 'center',
    alignItems: 'center',
  },
  plantInfoDiv: {
    width: '35%',
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  plantInfoText: {
    color: gray.iron(),
    fontSize: 10,
  },
  plantText: {
    backgroundColor: gray.shark(),
    flex: 0.375,
    width: width / 1.3,
    alignSelf: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  plantTextDescription: {
    color: gray.iron(),
    textAlign: 'left',
    padding: '4%',
    fontSize: 10,
  },
  containerBody: {
    flex: 0.6,
    backgroundColor: purple.frenchLilacLighter(),
    borderTopRightRadius: 20,
    borderTopLeftRadius: 20,
  },
  menuBar: {
    flexDirection: 'row',
    alignItems: 'center',
    alignSelf: 'center',
    justifyContent: 'space-around',
    width: width / 1.7,
  },
  menuBarTab: {
    width: width / 4,
    alignItems: 'center',
    justifyContent: 'center',
  },
  menuBarTabText: {
    color: gray.iron(),
    fontSize: 12,
  },
  bodyPlants: {
    width: width / 1.2,
    alignSelf: 'center',
    padding: 10,
    flex: 0.35,
  },
  bodyPlantsTitle: {
    color: gray.shark(),
    fontSize: 16,
    marginBottom: 10,
  },
  userDiv: {
    padding: 5,
    width: width / 4.5,
    alignItems: 'center',
  },
  UserImg: {
    width: width / 5,
    height: width / 5,
  },
  userName: {
    fontSize: 12,
    textAlign: 'center',
  },
  bodyTopics: {
    width: width / 1.2,
    alignSelf: 'center',
    padding: 0,
    flex: 0.65,
  },
  TopicDivContainer: {
    width: width / 1.2,
    alignSelf: 'center',
    marginVertical: '3%',
    height: height / 3.8,
  },
  TopicImg: {
    width: '100%',
    height: height / 3.8,
    alignSelf: 'center',
    flexDirection: 'column-reverse',
    alignItems: 'center',
  },
  TopicDescriptionDiv: {
    width: '100%',
    flex: 0.42,
    backgroundColor: green.darkFern(),
    opacity: 0.8,
    flexDirection: 'column',
    justifyContent: 'space-around',
    paddingLeft: 15,
  },
  TopicCommentsDiv: {
    width: '100%',
    flex: 0.18,
    backgroundColor: green.zuccini(),
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 15,
  },
  TopicUsername: {
    color: gray.iron(),
    fontWeight: '400',
    fontSize: 13,
  },
  TopicTitle: {
    color: gray.iron(),
    fontWeight: 'bold',
    fontSize: 13,
  },
  TopicDescription: {
    color: gray.iron(),
    fontWeight: '500',
    fontSize: 10,
  },

  ButtonBackground: {
    position: 'absolute',
    zIndex: 0,
    marginTop: height / 1.2,
    marginLeft: width / 1.35,
    width: width / 5,
    height: width / 5,
    borderRadius: 100,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: purple.lightWisteria(0.3),
  },
  ButtonDiv: {
    width: width / 6.8,
    height: width / 6.8,
    borderRadius: 100,
    backgroundColor: gray.shark(),
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default styles;
