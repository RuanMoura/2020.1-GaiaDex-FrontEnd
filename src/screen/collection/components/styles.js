import { StyleSheet, Dimensions } from 'react-native';

const { height, width } = Dimensions.get('screen');
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#F2E0F5',
  },
  header: {
    flex: 0.25,
    flexDirection: 'column',
    width: width / 1.1,
    alignSelf: 'center',
  },
  headerIcon: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'flex-start',
    flex: 0.6,
  },
  headerTitle: {
    width: width / 1.2,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  headerTitleText: {
    fontSize: 35,
    fontWeight: '600',
    paddingLeft: 5,
  },
  body: {
    flex: 0.7,
    alignItems: 'center',
    paddingBottom: 20,
  },
  plant: {
    flex: 0.45,
    maxHeight: height / 3,
    width: width / 1.2,
    flexDirection: 'row',
    marginBottom: '10%',
  },
  plantBackground: {
    flex: 1,
  },
  plantDescription: {
    flex: 1,
    backgroundColor: '#59A44D',
    fontWeight: '600',
    height: '50%',
    alignSelf: 'center',
    marginTop: '10%',
    borderTopRightRadius: 20,
    borderBottomRightRadius: 20,
    flexDirection: 'column',
    alignItems: 'center',
    padding: 10,
  },
  plantDescriptionTitle: {
    fontSize: 14,
    color: 'white',
    paddingBottom: 5,
    textAlign: 'center',
  },
  plantDescriptionText: {
    fontSize: 12,
    color: 'white',
    textAlign: 'center',
  },
  plantButtonDiv: {
    flexDirection: 'row',
    width: '100%',
    height: '50%',
    alignSelf: 'flex-start',
    margin: 5,
    marginTop: '4%',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  plantButton: {
    width: '45%',
    height: '80%',
    backgroundColor: 'orange',
    alignSelf: 'flex-start',
    margin: 5,
    marginTop: '5%',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  deleteButton: {
    width: '45%',
    height: '80%',
    backgroundColor: 'red',
    alignSelf: 'flex-start',
    margin: 5,
    marginTop: '5%',
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  deleteButtonText: {
    fontSize: 12,
    color: 'white',
  },
});
export default styles;
