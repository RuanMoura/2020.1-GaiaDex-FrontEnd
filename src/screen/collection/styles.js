import { StyleSheet, Dimensions } from 'react-native';
import { gray, highlight, purple } from '../../theme/colorPalette';

const { height, width } = Dimensions.get('screen');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: purple.frenchLilacLighter(),
    // paddingTop: Constants.statusBarHeight,
  },
  editView: {
    backgroundColor: purple.frenchLilacLighter(),
    width: (width * 4) / 5,
    height: (height * 2) / 9,
    marginTop: (height * 5) / 36,
    marginLeft: width / 10,
    borderRadius: 10,
    shadowColor: gray.darkest(),
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 50,
    paddingHorizontal: width / 25,
    paddingVertical: height / 90,
    justifyContent: 'space-between',
  },
  editTitle: {
    fontSize: 17,
  },
  editTextInput: {
    height: 40,
    width: (width * 4) / 5 - (width * 2) / 25,
    borderBottomColor: purple.grapeDark(),
    borderBottomWidth: 1,
  },
  editButton: {
    fontSize: 14,
    color: highlight.lochmara(),
    marginLeft: (width * 9) / 125,
  },
  exploreContainer: {
    flexDirection: 'row',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 0.22,
    justifyContent: 'space-between',
    alignContent: 'center',
    alignItems: 'center',
  },
  exploreText: {
    fontSize: 40,
    fontWeight: '600',
    color: gray.outerSpace(),
  },
  popularContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 0.7,
    justifyContent: 'flex-start',
  },
  tabContainer: {
    flexDirection: 'row',
    width: width / 1.7,
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: '10%',
  },
  tabText: {
    fontSize: 18,
    fontWeight: '400',
  },
  tabTextactivate: {
    fontSize: 18,
    fontWeight: '500',
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: gray.darkest(),
  },
  popularTabs: {
    flexDirection: 'row',
    flex: 0.2,
  },
  popularTab: {
    fontSize: 22,
    fontWeight: '400',
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: gray.darkest(),
  },
  popularContent: {
    flex: 0.8,
  },
  FavoritePlant: {
    width: width / 3,
    flex: 1,
    flexDirection: 'column-reverse',
    borderRadius: 30,
    marginRight: 15,
    backgroundColor: purple.grapeDark(0.9),
  },
  FavoritePlantImg: {
    width: '100%',
    resizeMode: 'cover',
    flexDirection: 'column-reverse',
    flex: 1,
    justifyContent: 'space-between',
  },
  plantInfo: {
    backgroundColor: purple.grapeDark(0.9),
    flex: 0.3,
    borderBottomRightRadius: 30,
    borderBottomLeftRadius: 30,
    width: width / 3,
    alignSelf: 'center',
  },
  myPlantsContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 1,
    justifyContent: 'flex-start',
  },
  myPlantsText: {
    paddingTop: 30,
    color: gray.outerSpace(),
    fontSize: 26,
    fontWeight: '600',
  },
  myplantPlant: {
    marginTop: 10,
    width: width / 2.3,
    height: height / 4,
    flexDirection: 'column-reverse',
    borderRadius: 20,
    marginRight: 15,
    backgroundColor: 'brown',
  },
  myplantInfo: {
    backgroundColor: purple.grapeDark(0.9),
    flex: 0.3,
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    width: width / 2.3,
    alignItems: 'center',
    justifyContent: 'space-around',
    flexDirection: 'row',
  },
  plantText: {
    padding: 5,
    fontSize: 10,
    textAlign: 'center',
    color: purple.frenchLilacLighter(),
  },
});
export default styles;
