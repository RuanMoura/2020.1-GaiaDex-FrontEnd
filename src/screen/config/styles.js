import { StyleSheet, Dimensions } from 'react-native';

const largura = Dimensions.get('screen').width;
const altura = Dimensions.get('screen').height;

const styles = StyleSheet.create({
  containerView: {
    flex: 1,
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  containerHeader: {
    flex: 0.2,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: largura,
    paddingRight: '8%',
  },
  containerHeaderIcon: {
    flexDirection: 'row',
  },
  headerTitle: {
    fontSize: 32,
    color: '#242528',
  },
  notificationContainer: {
    flex: 0.7,
  },
  notification: {
    flex: 0.1,
    flexDirection: 'row',
    marginVertical: '3%',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: largura / 1.2,
  },
  notificationText: {
    fontSize: 15,
    fontWeight: '500',
    color: '#242528',
  },
});

export default styles;
