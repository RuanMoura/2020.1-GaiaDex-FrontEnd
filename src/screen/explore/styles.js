import { StyleSheet, Dimensions } from 'react-native';

import Constants from 'expo-constants';

const largura = Dimensions.get('screen').width;
const altura = Dimensions.get('screen').height;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#F2E0F5',
    paddingTop: Constants.statusBarHeight,
  },
  exploreContainer: {
    flexDirection: 'row',
    alignSelf: 'center',
    width: largura / 1.1,
    flex: 0.22,
    justifyContent: 'space-between',
    alignContent: 'center',
    alignItems: 'center',
  },
  exploreText: {
    fontSize: 40,
    fontWeight: '600',
    color: '#31353A',
  },
  popularContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: largura / 1.1,
    flex: 0.35,
    justifyContent: 'flex-start',
  },
  popularTabs: {
    flexDirection: 'row',
    flex: 0.2,
  },
  popularTab: {
    fontSize: 22,
    fontWeight: '400',
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: '#000',
  },
  popularContent: {
    flex: 0.8,
  },
  FavoritePlant: {
    width: largura / 2.7,
    height: altura / 4,
    flexDirection: 'column-reverse',
    borderRadius: 30,
    marginRight: 15,
    backgroundColor: 'rgba(52, 9, 37, 0.91)',
  },
  FavoritePlantImg: {
    width: '100%',
    resizeMode: 'cover',
    flexDirection: 'column-reverse',
    flex: 1,
  },
  plantInfo: {
    backgroundColor: 'rgba(52, 9, 37, 0.91)',
    flex: 0.2,
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    width: largura / 2.7,
    alignSelf: 'center',
  },
  myPlantsContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: largura / 1.1,
    flex: 0.3,
    justifyContent: 'flex-start',
  },
  myPlantsText: {
    paddingTop: 30,
    color: '#31353A',
    fontSize: 26,
    fontWeight: '600',
  },
  myplantPlant: {
    marginTop: 10,
    height: altura / 6,
    width: largura / 4.5,
    flexDirection: 'column-reverse',
    borderRadius: 20,
    marginRight: 15,
    backgroundColor: 'brown',
  },
  myplantInfo: {
    backgroundColor: 'rgba(52, 9, 37, 0.91)',
    flex: 0.3,
    borderBottomRightRadius: 10,
    borderBottomLeftRadius: 10,
    width: largura / 4.5,
    alignSelf: 'center',
  },
  plantText: {
    padding: 5,
    textAlign: 'center',
    fontSize: 10,
    color: 'white',
  },
});
export default styles;
