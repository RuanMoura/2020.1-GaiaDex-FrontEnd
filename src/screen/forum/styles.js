import { StyleSheet, Dimensions } from 'react-native';
import Constants from 'expo-constants';
import { gray, green, purple } from '../../theme/colorPalette';

const { height, width } = Dimensions.get('screen');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: purple.frenchLilacLighter(),
    paddingTop: Constants.statusBarHeight,
  },
  forumContainer: {
    flexDirection: 'row',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 0.2,
    justifyContent: 'space-between',
    alignContent: 'center',
    alignItems: 'center',
  },
  forumText: {
    fontSize: 40,
    fontWeight: '600',
    color: gray.outerSpace(),
  },
  myforumsContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 0.28,
    justifyContent: 'flex-start',
  },
  myforumsTabs: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    flex: 0.2,
    paddingRight: 10,
  },
  myforumsTab: {
    fontSize: 22,
    fontWeight: '400',
    color: gray.outerSpace(),
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: gray.shark(),
  },
  myforumsContent: {
    flex: 0.8,
  },
  FavoritePlant: {
    width: width / 4,
    height: width / 3.5,
    flexDirection: 'column',
    marginRight: 15,
    marginTop: 15,
  },
  FavoriteTopicImg: {
    flex: 0.95,
    width: '100%',
    alignSelf: 'center',
    borderRadius: 60,
    backgroundColor: gray.shark(),
  },
  FavoriteTopicText: {
    fontSize: 8,
    color: gray.outerSpace(),
    fontWeight: 'bold',
    textAlign: 'center',
  },
  plantInfo: {
    backgroundColor: purple.grapeDark(0.9),
    borderBottomRightRadius: 30,
    borderBottomLeftRadius: 30,
    width: width / 2.7,
    alignSelf: 'center',
  },
  plantText: {
    padding: 5,
    fontSize: 13,
    color: gray.iron(),
  },
  topPostContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    width: width / 1.1,
    flex: 0.4,
    justifyContent: 'flex-end',
  },
  topPostTabs: {
    width: width / 2,
    alignSelf: 'flex-start',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  topPostText: {
    paddingTop: 16,
    color: gray.outerSpace(),
    fontSize: 16,
    fontWeight: '400',
  },
  topPostTextactive: {
    paddingTop: 16,
    color: gray.outerSpace(),
    fontSize: 18,
    fontWeight: '600',
    textDecorationLine: 'underline',
    textDecorationStyle: 'solid',
    textDecorationColor: gray.shark(),
  },
  topPostPlant: {
    marginTop: 15,
    height: height / 3.6,
    width: width / 1.1,
    flexDirection: 'column-reverse',
    borderRadius: 30,
  },
  topPostInfo: {
    backgroundColor: green.jewel(0.9),
    flex: 0.6,
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    width: width / 1.1,
    alignSelf: 'center',
  },
  topPostBack: {
    borderRadius: 20,
    marginVertical: 10,
    backgroundColor: green.jewel(0.9),
    flex: 1,
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    width: width / 1.1,
    alignSelf: 'center',
  },
  postTitle: {
    padding: 5,
    flex: 0.2,
    fontSize: 16,
    color: gray.iron(),
  },
  postDescription: {
    padding: 5,
    flex: 0.2,
    fontSize: 12,
    color: gray.iron(),
  },
  postData: {
    padding: 6,
    flex: 0.2,
    fontSize: 8,
    color: gray.iron(),
  },
  posStatus: {
    flex: 0.7,
    paddingHorizontal: 15,
    flexDirection: 'row',
    backgroundColor: green.zuccini(0.95),
    borderBottomRightRadius: 20,
    borderBottomLeftRadius: 20,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  postLikes: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
});
export default styles;
