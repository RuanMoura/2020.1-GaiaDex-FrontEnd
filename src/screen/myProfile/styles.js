import { StyleSheet, Dimensions } from "react-native"
import Constants from "expo-constants";

const altura = Dimensions.get("screen").height;
const largura = Dimensions.get("screen").width;

const styles = StyleSheet.create({
    framePerfil:{
        flex: 1,
        backgroundColor: "#242528",
        paddingTop: Constants.statusBarHeight,
        //justifyContent: "center",
    },
    perfilTextView: {
        alignItems:"center",
        flexDirection: "row",
        justifyContent: "space-between",
        alignSelf: "center",
        width: largura/1.2,
    },
    perfilText: {
        fontSize: altura/20,
        color: "#D6DADF",
        fontWeight: "500",
    },
    vector:{
        // alignSelf:"center",
        marginTop: 10,
        height: altura*4/11,
        width: largura,
    },
    photoView:{
        backgroundColor: "white",
        height: largura*3/8,
        width: largura*3/8,
        borderRadius: largura*3/16,
        marginLeft: largura/2 - largura*3/16,
        marginTop: altura*2/11 - largura*3/16 - largura*1/8,
    },
    name:{
        color: "#E5E5E5",
        fontSize: 13,
        textAlign: "center",
        marginTop: 7,
    },
    frameDown:{
        flex: 1,
        backgroundColor: "#F2E0F5",
    },
    sumary:{
        position: "absolute",
        backgroundColor: "#19BB53",
        width: largura*10/11,
        height: 114,
        marginHorizontal: (largura - largura*10/11)/2,
        marginVertical: (altura/2)-57,
        borderRadius: 20,
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center",
    },
    sumaryComponents:{
        height: 100,
        width: largura*10/33,
        paddingTop: 25,
        alignItems: "center",
    },
    minhasAtividades:{
        marginTop: 75,
        marginLeft: (largura - largura*10/11)/2,
        fontSize: 18,
        fontWeight: "600",
    },
});

export default styles;