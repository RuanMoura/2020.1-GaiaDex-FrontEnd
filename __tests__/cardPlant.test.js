import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

import PlantCard from '../src/screen/cardPlant/cardPlant';
import styles from '../src/screen/cardPlant/styles';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

import renderer from 'react-test-renderer';


const wrapper = mount(<PlantCard navigation={{ getParam: jest.fn() }} />);

describe('<TopicCreate />', () => {
  it('renders everything', () => {
    const tree = renderer
      .create(<PlantCard navigation={{ getParam: jest.fn() }}></PlantCard>)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(wrapper.find('View').length).toBe(40);
    expect(wrapper.find('Header').length).toBe(1);
    expect(wrapper.find('ImageBackground').length).toBe(1);
    expect(wrapper.find('Text').length).toBe(30);
    expect(wrapper.find('TouchableOpacity').length).toBe(3);
    expect(wrapper.find('PlantCardTopic').length).toBe(1);
    expect(wrapper.find('NewTopic').length).toBe(1);
  });
  it('renders View correctly', () => {
    expect(wrapper.find('View').at(0)).toHaveStyle(styles.containerView);
    expect(wrapper.find('View').at(6)).toHaveStyle(styles.containerPlant);
    expect(wrapper.find('View').at(10)).toHaveStyle(styles.plantInfo);
    expect(wrapper.find('View').at(12)).toHaveStyle(styles.plantInfoStats);
    expect(wrapper.find('View').at(14)).toHaveStyle(styles.plantInfoDiv);
    expect(wrapper.find('View').at(16)).toHaveStyle(styles.plantInfoDiv);
    expect(wrapper.find('View').at(18)).toHaveStyle(styles.plantText);
    expect(wrapper.find('View').at(20)).toHaveStyle(styles.menuBar);
  });
  it('renders Header correctly', () => {
    expect(wrapper.find('Header').at(0)).toHaveProp('navigation');
  });
  it('renders Text correctly', () => {
    expect(wrapper.find('Text').at(2)).toHaveStyle(styles.plantInfoTitle);
    expect(wrapper.find('Text').at(6)).toHaveStyle(styles.plantInfoText);
    expect(wrapper.find('Text').at(6)).toHaveText(' Usuários');
    expect(wrapper.find('Text').at(10)).toHaveStyle(styles.plantInfoText);
    expect(wrapper.find('Text').at(10)).toHaveText(' Tópicos');

    expect(wrapper.find('Text').at(14)).toHaveStyle(styles.plantTextDescription);
    expect(wrapper.find('Text').at(14)).toHaveText('Esta é uma comunidade certificada/padrão do app. Direcionada para se interagir sobre quaisquer assuntos envolvendo cenouras.');

    expect(wrapper.find('Text').at(16)).toHaveStyle(styles.menuBarTabText);
    expect(wrapper.find('Text').at(16)).toHaveText('Informações');
    expect(wrapper.find('Text').at(20)).toHaveStyle(styles.menuBarTabText);
    expect(wrapper.find('Text').at(20)).toHaveText('Topicos');
  });
  it('renders TouchableOpacity correctly', () => {
    expect(wrapper.find('TouchableOpacity').at(0)).toHaveStyle(
      styles.menuBarTab
    );
    expect(wrapper.find('TouchableOpacity').at(0)).toHaveProp('onPress');
    expect(wrapper.find('TouchableOpacity').at(1)).toHaveStyle(
      styles.menuBarTab
    );
    expect(wrapper.find('TouchableOpacity').at(1)).toHaveProp('onPress');
  });
  it('renders PlantCardTopic correctly', () => {
    expect(wrapper.find('PlantCardTopic').at(0)).toHaveProp('navigation');
    expect(wrapper.find('PlantCardTopic').at(0)).toHaveProp('plantID');
  });
});
