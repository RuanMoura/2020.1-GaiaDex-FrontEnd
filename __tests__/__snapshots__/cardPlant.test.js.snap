// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopicCreate /> renders everything 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#242528",
      "flex": 1,
      "flexDirection": "column",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#242528",
        "flex": 0.1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      }
    }
  >
    <Text />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#242528",
        "flex": 0.3,
        "flexDirection": "column",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignSelf": "center",
          "flex": 0.4,
          "marginTop": "-9%",
          "width": 357.1428571428571,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": 357.1428571428571,
            },
            Object {
              "borderRadius": 25,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "backgroundColor": "#242528",
          "flex": 0.25,
          "flexDirection": "column",
          "width": 576.9230769230769,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#D6DADF",
            "flex": 0.6,
            "fontSize": 23,
          }
        }
      />
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 0.4,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": 576.9230769230769,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
              "width": "35%",
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#D6DADF",
                "fontSize": 10,
              }
            }
          >
             Usuários
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-around",
              "width": "35%",
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#D6DADF",
                "fontSize": 10,
              }
            }
          >
             Tópicos
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#242528",
          "flex": 0.375,
          "flexDirection": "row",
          "justifyContent": "center",
          "width": 576.9230769230769,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#D6DADF",
            "fontSize": 10,
            "padding": "4%",
            "textAlign": "left",
          }
        }
      >
        Esta é uma comunidade certificada/padrão do app. Direcionada para se interagir sobre quaisquer assuntos envolvendo cenouras.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "width": 441.1764705882353,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
            "width": 187.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 12,
            }
          }
        >
          Informações
        </Text>
        <Text />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
            "width": 187.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 12,
            }
          }
        >
          Topicos
        </Text>
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F9F6F4",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "flex": 0.6,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "flex": 0.35,
          "padding": 10,
          "width": 625,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242528",
            "fontSize": 16,
            "marginBottom": 10,
          }
        }
      >
        Contribuidores:
      </Text>
      <RCTScrollView
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flex": 0.65,
          "padding": 0,
          "width": 625,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242528",
            "fontSize": 16,
            "marginBottom": 10,
          }
        }
      >
        Tópicos relacionados:
      </Text>
      <RCTScrollView
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 0, 0, 0.3)",
        "borderRadius": 100,
        "height": 150,
        "justifyContent": "center",
        "marginLeft": 555.5555555555555,
        "marginTop": 1111.6666666666667,
        "position": "absolute",
        "width": 150,
        "zIndex": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#242528",
          "borderRadius": 100,
          "height": 110.29411764705883,
          "justifyContent": "center",
          "opacity": 1,
          "width": 110.29411764705883,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
